### PR TITLE
permisson user

### PR DESCRIPTION
**Pull request by phamkykhoi**

### CODE REVIEW

File: app/models/role.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Role < ActiveRecord::Base
+  has_and_belongs_to_many :users, :join_table => :users_roles
```
Comment (dieunb) [inline app/models/role.rb]: coding convention

File: app/models/role.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  belongs_to :resource, :polymorphic => true
```
Comment (dieunb) [inline app/models/role.rb]: coding convention

File: app/models/role.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+  validates :resource_type,
+            :inclusion => {:in => Rolify.resource_types},
```
Comment (dieunb) [inline app/models/role.rb]: coding convention

File: app/models/role.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+            :allow_nil => true
```
Comment (dieunb) [inline app/models/role.rb]: coding convention

File: app/helpers/sessions_helper.rb [hunk 5]
```diff
@@ -1,53 +0,0 @@
-module SessionsHelper
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: checkout this file

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -4,25 +4,13 @@ class ApplicationController < ActionController::Base
...
-    end
-  end
-
   def login_redirect_admin
-    unless logged_in? && is_admin?
-      flash[:danger] = "You don't have permission to access on system"
-      redirect_to root_path
-    end
+    flash[:danger] = "You don't have permission to access admin system"
+    redirect_to root_path if user.has_role? :admin
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: if user is not admin, what is action?

File: app/models/role.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+class Role < ActiveRecord::Base
+  has_and_belongs_to_many :users, :join_table :users_roles
+  belongs_to :resource, :polymorphic :true
+
+  validates :resource_type,
+            :inclusion => {:in => Rolify.resource_types},
```
Comment (dieunb) [inline app/models/role.rb]: coding convention

File: app/models/role.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+class Role < ActiveRecord::Base
+  has_and_belongs_to_many :users, :join_table :users_roles
```
Comment (dieunb) [inline app/models/role.rb]: don't use has_and_belongs_to_many.

File: app/models/role.rb [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+  belongs_to :resource, :polymorphic :true
+
+  validates :resource_type,
+            :inclusion => {:in => Rolify.resource_types},
+            :allow_nil :true
+
+  scopify
+end
```
Comment (dieunb) [inline app/models/role.rb]: add new line at end of file

File: app/controllers/application_controller.rb [hunk 10]
```diff
@@ -2,27 +2,17 @@ class ApplicationController < ActionController::Base
...
-      redirect_to login_path
-    end
-  end
-
+  protected
   def login_redirect_admin
-    unless logged_in? && is_admin?
-      flash[:danger] = "You don't have permission to access on system"
+    if user_signed_in? && !current_user.has_role?(:admin) 
+      flash[:danger] = "You don't have permission to access admin system"
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: i18n

File: app/controllers/application_controller.rb [hunk 11]
```diff
@@ -2,27 +2,17 @@ class ApplicationController < ActionController::Base
...
     end
   end
 
-  def logged_in_user
-    unless logged_in?
-      flash[:danger] = "Please log in take access system"
-      redirect_to login_path
-    end
+  def login_redirect
+    flash[:danger] = "Please login"
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: i18n

File: config/initializers/rolify.rb [hunk 12]
```diff
@@ -0,0 +1,7 @@
+Rolify.configure do |config|
+  # By default ORM adapter is ActiveRecord. uncomment to use mongoid
+  # config.use_mongoid
+
+  # Dynamic shortcuts for User class (user.is_admin? like methods). Default is: false
+  # config.use_dynamic_shortcuts
+end
```
Comment (dieunb) [inline config/initializers/rolify.rb]: add new line at end of file

File: config/routes.rb [hunk 13]
```diff
@@ -64,12 +64,9 @@
...
   # delete 'logout' => 'sessions#destroy'
 
-  # namespace :admin do
-  #   root :to => 'dashbroads#home'
-  #   resources :categories
-  #   resources :words
-  #   resources :users
-  # end
+  namespace :admin do
+    root :to => 'dashbroads#home'
```
Comment (dieunb) [inline config/routes.rb]: it is rails 3 syntax, pls using rails 4 syntax

File: app/controllers/application_controller.rb [hunk 14]
```diff
@@ -2,27 +2,19 @@ class ApplicationController < ActionController::Base
...
-      redirect_to login_path
-    end
-  end
-
+  protected
   def login_redirect_admin
-    unless logged_in? && is_admin?
-      flash[:danger] = "You don't have permission to access on system"
+    if user_signed_in? && !current_user.has_role?(:admin) 
+      flash[:danger] = I18n.t 'notice.danger'
```
Comment (nguyenduyta) [inline app/controllers/application_controller.rb]: ' => "

File: app/controllers/application_controller.rb [hunk 15]
```diff
@@ -2,27 +2,19 @@ class ApplicationController < ActionController::Base
...
     end
   end
 
-  def logged_in_user
-    unless logged_in?
-      flash[:danger] = "Please log in take access system"
-      redirect_to login_path
+  def login_redirect
+    if !user_signed_in? 
+      flash[:danger] = I18n.t 'notice.danger'
```
Comment (nguyenduyta) [inline app/controllers/application_controller.rb]: nt

File: app/controllers/application_controller.rb [hunk 16]
```diff
@@ -2,27 +2,19 @@ class ApplicationController < ActionController::Base
...
       redirect_to root_path
     end
   end
 
-  def logged_in_user
-    unless logged_in?
-      flash[:danger] = "Please log in take access system"
-      redirect_to login_path
+  def login_redirect
+    if !user_signed_in?
```
Comment (nguyenduyta) [inline app/controllers/application_controller.rb]: chuyển if thành unless

File: config/routes.rb [hunk 17]
```diff
@@ -64,12 +64,9 @@
...
   # delete 'logout' => 'sessions#destroy'
 
-  # namespace :admin do
-  #   root :to => 'dashbroads#home'
-  #   resources :categories
-  #   resources :words
-  #   resources :users
-  # end
+  namespace :admin do
+    root 'dashbroads#home'
```
Comment (nguyenduyta) [inline config/routes.rb]: ' => "

**Pull request merged**